src/bitcoind.cpp

```diff
@@ -50,11 +50,9 @@ static bool AppInit(int argc, char* argv[])
 
     util::ThreadSetInternalName("init");
 
-    //
-    // Parameters
-    //
     // If Qt is used, parameters/bitcoin.conf are parsed in qt/bitcoin.cpp's main()
     SetupServerArgs(node);
+    ArgsManager& args = *Assert(node.args);
```

Comment:
In commit "init: Pass reference to ArgsManager around instead of relying on global" (fa40017706e08b4de111e8e57aabeced60881a57)

No need to change here, but it'd be nice for a few of my pr's if SetupServerArgs took an `ArgsManager` parameter instead of a `NodeContext` parameter. (#10102 does this, but it's nice to shrink that PR.) It'd make this code a little more straightforward too, replacing the assertion with an assignment.

Reply:
I'll leave that for later because I don't understand how passing a node context with an args pointer is different to passing an args reference. The cross-process `setupServerArgs` (lowercase `s`) doesn't pass anything, so this seems unrelated to multiprocess, no?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/19779#discussion_r475777703

> I'll leave that for later because I don't understand how passing a node context with an args pointer is different to passing an args reference. The cross-process `setupServerArgs` (lowercase `s`) doesn't pass anything, so this seems unrelated to multiprocess, no?

That's fine. To answer the question about multiprocess, #10102 changes SetupServerArgs not to take a NodeContext because the gui process also calls SetupServerArgs directly to register arguments, and be able to call gArgs.GetChainName() and gArgs.GetArg() to get values like -prune, -proxy, -datadir, etc. Also https://github.com/bitcoin-core/gui/pull/35 removes the lowercase `setupServerArgs` function.

We should just keep in mind trying not to overuse NodeContext, because passing NodeContext around can obscure what functions are doing it make them harder to call in places where NodeContext isn't available. If a function like SetupServerArg only uses one member of the NodeContext struct, probably better just to pass that one member instead of the whole struct.

Reply:
Ah, I see. I think this can be changed nicely in one go when the type of node.args is changed from raw ptr to unique_ptr in a follow-up.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/19779#discussion_r475713788

> it'd be nice for a few of my pr's if SetupServerArgs took an ArgsManager parameter

FWIW, this is now done earlier in the latest version of https://github.com/bitcoin/bitcoin/pull/19160#pullrequestreview-464492390 ([compare](https://github.com/ryanofsky/bitcoin/compare/pr/ipc-echo.11-rebase..pr/ipc-echo.12))